sympy/core/expr.py

```diff
@@ -866,7 +866,7 @@ def _eval_is_extended_positive_negative(self, positive):
 
             # check to see that we can get a value
             try:
-                n2 = self._eval_evalf(2)
+                n2 = self._eval_evalf(3)
```

Comment:
We should be trying to get rid of this. Why has it changed to 3?

Reply:
Actually, I tried getting rid of that specific `_eval_evalf` call and making the flow continue directly into `evalf()` (#L880) however overall it tended to make sympy slow (the online test suite of github took more than 1-2 hours to evaluate for each test ). 

So I don't think it's optimal to get rid of that unless we find a way to eliminate both the `_eval_evalf()` and `evalf` call from this function without compromising that performance, essentially removing the numerical evaluation completely.  But as mentioned here https://github.com/sympy/sympy/pull/20668#issuecomment-751800149 right now there doesn't seem to be  any efficient way to resolve that without numerical evaluation.

It has been changed to 3 to increase precision as cases like these https://github.com/sympy/sympy/pull/20668#issuecomment-751744607 might get evaluated as insignificant and return None.

Reply:
> unless we find a way to eliminate both the `_eval_evalf()` and `evalf`

I agree that there's no point removing one without the other. Ultimately I think that these calls will need to be there for *some* cases but the idea would be:

1. To implement other ways so that numerical evaluation is usually not needed.
2. To make it safe to use numerical evaluation even for very large numbers by having it not evaluate in cases where the result would be extremely large.

The approach I would take with this is to try removing both calls and then try to add in other handlers for `is_le` etc that can replace them as much as possible and in that way to fix as many tests as possible that break. Then we can add the `evalf` calls back but in such a way that they are not used in simple cases which is point 1.